packages/editor-ui/src/components/WorkflowSelectorParameterInput/useWorkflowResourcesLocator.ts

```diff
@@ -0,0 +1,92 @@
+import { ref, computed } from 'vue';
+import { useWorkflowsStore } from '@/stores/workflows.store';
+import { sortBy } from 'lodash-es';
+import type { Router } from 'vue-router';
+import { VIEWS } from '@/constants';
+
+import type { IWorkflowDb } from '@/Interface';
+
+export function useWorkflowResourcesLocator(router: Router) {
+	const workflowsStore = useWorkflowsStore();
+	const workflowsResources = ref<Array<{ name: string; value: string; url: string }>>([]);
+	const isLoadingResources = ref(true);
+	const searchFilter = ref('');
+	const PAGE_SIZE = 40;
+
+	const sortedWorkflows = computed(() =>
+		sortBy(workflowsStore.allWorkflows, (workflow) =>
+			new Date(workflow.updatedAt).valueOf(),
+		).reverse(),
+	);
+
+	const hasMoreWorkflowsToLoad = computed(
+		() => workflowsStore.allWorkflows.length > workflowsResources.value.length,
+	);
+
+	const filteredResources = computed(() => {
+		if (!searchFilter.value) return workflowsResources.value;
+
+		return workflowsStore.allWorkflows
+			.filter((resource) => resource.name.toLowerCase().includes(searchFilter.value.toLowerCase()))
+			.map(workflowsMapper);
+	});
+
+	async function populateNextWorkflowsPage() {
+		if (workflowsStore.allWorkflows.length <= 1) {
+			await workflowsStore.fetchAllWorkflows();
+		}
+		const nextPage = sortedWorkflows.value.slice(
+			workflowsResources.value.length,
+			workflowsResources.value.length + PAGE_SIZE,
+		);
+
+		workflowsResources.value.push(...nextPage.map(workflowsMapper));
+	}
+
+	async function setWorkflowsResources() {
+		isLoadingResources.value = true;
+		await populateNextWorkflowsPage();
+		isLoadingResources.value = false;
+	}
+
+	function workflowsMapper(workflow: IWorkflowDb) {
```

Comment:
I was looking at this function name and didn't really understand what it would do from the name. Could we make this a name like `workflowToResource` or, since it's an internal function, we could use `_` as the first character to indicate so? Either option would work for me, because now I was a bit confused